### PR TITLE
support Instant in DataObject converters

### PR DESCRIPTION
**Pull request by jo5ef**
This adds `DataObject` converter support for `java.time.Instant` properties.

### CODE REVIEW

**Comment by vietj** [comment]
this looks great however we need this to be also supported in various languages that perform data object conversion so it is properly supported

**Comment by vietj** [comment]
well actually I think that only Kotlin would need support for it

**Comment by vietj** [comment]
can you add support for instant in the data object TCK that is used by other projects ?

**Comment by jo5ef** [comment]
Ok, I've added `Instant` to all remaining dataobject test objects and also added them to vertx-lang-kotlin (see [this PR](https://github.com/vert-x3/vertx-lang-kotlin/pull/63)). I haven't checked the other lang projects yet but I can and if you're interested to merge this I can also update the docs before you do.

**Comment by vietj** [comment]
I'm interested, I think you can have a look at Ceylon too that generates data object equivalents, at least check it compiles

**Comment by vietj** [comment]
we need actually to check it works for scala, ruby, js, groovy and ceylon because they use the tck

**Comment by jo5ef** [comment]
So, I had a look at the other projects:

* kotlin - https://github.com/vert-x3/vertx-lang-kotlin/pull/63
* scala - https://github.com/vert-x3/vertx-lang-scala/pull/70
* ruby - build fails, but not because of changes in this PR
* js - build success
* groovy - build success
* ceylon - https://github.com/vert-x3/vertx-lang-ceylon/pull/28

I never use any of the above languages, so I hope my changes make sense.

**Comment by vietj** [comment]
I will look at the ruby failure and have the other reviewed

**Comment by vietj** [comment]
@jo5ef the vertx-lang-ruby build has been fixed, you can have a look again

**Comment by jo5ef** [comment]
@vietj thanks, ruby also builds successfully with this change

**Comment by vietj** [comment]
@jo5ef `JsonObject` and `JsonArray` supports already the `Instant` type so you need instead to use that instead of doing the conversion in the converter

**Comment by vietj** [comment]
looking at it, I think we should aim to not add a new codegen type for this purpose and instead use `OTHER` that already exists. Then in the `DataObjectModel` when it is `OTHER` type we should make available the name of the class and then the converter should handle it this way.

This way it's easier to support more converted types in the future.

**Comment by jo5ef** [comment]
@vietj  I would have used the `getInstant` and `put` methods from `JsonObject` but the converter generates a `Iterable<java.util.Map.Entry<String, Object>>` parameter in the `fromJson` method so I couldn't and I did not want to change that signature.
Using `OTHER` instead of a new `INSTANT` type works as well, I can rewrite if you like.

**Comment by vietj** [comment]
yes please rewrite, we will actually support more OTHER types in 3.6 so this fits well the purpose

**Comment by vietj** [comment]
see https://github.com/vert-x3/wiki/wiki/Vert.x-Roadmap#codegen-enhanced-methods

**Comment by jo5ef** [comment]
ok, I rewrote to use OTHER and also updated the scala and ceylon PRs accordingly. kotlin didn't need any changes.

**Comment by vietj** [comment]
great I'll do another review.

@sammers21 is on contributing the generalisation of `OTHER` in codegen and is interested in following this effort as well

**Comment by codepitbull** [comment]
Nice work @jo5ef ! Waiting for this brnach to be merged, then I will merge the PR on vertx-lang-scala.

File: README.md [hunk 1]
```diff
@@ -370,6 +370,7 @@ Data object converter can be generated with `@DataObject(generateConverter=true)
 
 * the specific `io.vertx.core.Buffer` type
 * the set _`Basic`_
+* the `java.time.Instant` type
```
Comment (vietj) [inline README.md]: you should group that with the specific buffer type:

```
* these specific types
  * `io.vertx.core.Buffer`
  * `java.time.Instant`
```

**Comment by jo5ef** [comment]
ok, done. but cheatsheet generation (in general) doesn't seem to work atm

**Comment by vietj** [comment]
any idea why it doesn't work ?

**Comment by jo5ef** [comment]
yes, I guess [because of this](https://github.com/vert-x3/vertx-codegen/pull/197)

**Comment by vietj** [comment]
I'll have a look at it.

**Pull request merged**